test/native/setup-after-env.js

```diff
@@ -7,3 +12,9 @@ import { toBeVisible } from './matchers/to-be-visible';
 expect.extend( {
 	toBeVisible,
 } );
+
+// Automatically cleanup mounted React trees after rendering components with the React
+// Native Testing Library. Although this is already done by the library, we provide our
+// own logic in order to prevent unexpected "act" warnings after the test finishes.
+// Reference: https://callstack.github.io/react-native-testing-library/docs/api#cleanup
+afterEach( cleanup );
```

Comment:
I interpret this custom implementation of clean up to effectively intentionally ignore valid `act` warnings, as the clean up now no longer includes `await flushMicroTasks()`. I.e with this change, we make ourselves vulnerable to introducing additional `act` warnings. Additionally, the current `act` warnings may produce false positive tests in the "right" context (not sure what that hypothetical context might be at this moment given we do not fully understand the source of the `act` warnings). Is that correct? 

I understand why we may want to land this temporary solution, but I wanted to clarify the intent. I also wonder if we should reframe the PR title and description to explicitly state we are disabling end-of-test `act` warnings rather than "fixing" them. The explicitness may prove useful to future readers. WDYT?

Reply:
> I interpret this custom implementation of clean up to effectively intentionally ignore valid `act` warnings, as the clean up now no longer includes `await flushMicroTasks()`. I.e with this change, we make ourselves vulnerable to introducing additional `act` warnings. Additionally, the current `act` warnings may produce false positive tests in the "right" context (not sure what that hypothetical context might be at this moment given we do not fully understand the source of the `act` warnings). Is that correct?

Yes, that's correct. Actually, this solution diverges from the original idea of waiting for all resolvers during the editor initialization. As I commented in [this comment](https://github.com/WordPress/gutenberg/pull/38336#issuecomment-1025642939), it's not totally clear to me whether omitting the React tree updates made by those selectors could really affect the result of the tests. In any case, I think a deeper investigation might help us on both having a better context of what happens during initialization and getting clues on how to provide a better workaround 👍 .

> I understand why we may want to land this temporary solution, but I wanted to clarify the intent. I also wonder if we should reframe the PR title and description to explicitly state we are disabling end-of-test `act` warnings rather than "fixing" them. The explicitness may prove useful to future readers. WDYT?

Sure, I'll rename it to make it clearer.